react/src/lib/components/DeckGLMap/components/DistanceScale.tsx

```diff
@@ -1,26 +1,52 @@
 import React from "react";
 
 interface scaleProps {
+    // Needed the zoom value to calculate width in units
     zoom: number;
+    // Scale increment value
+    incrementValue: number;
+    // width per unit value to calculate width in units
+    widthPerUnit: number;
 }
-const DistanceScale: React.FC<scaleProps> = ({ zoom }: scaleProps) => {
-    const [width] = React.useState(100);
-    const pixelPerUnit = width / Math.pow(2, zoom);
-    const scaleBarStyle = {
-        width: width,
+
+const roundOffvalues = function (num: number, step: number) {
+    return Math.floor(num / step + 0.5) * step;
+};
+
+const DistanceScale: React.FC<scaleProps> = ({
+    zoom,
+    incrementValue,
+    widthPerUnit,
+}: scaleProps) => {
+    const [rulerWidth, setRulerWidth] = React.useState<number>(widthPerUnit);
```

Comment:
This should probably be initialized to 0, since it will be rewritten immediately anyway.